src/sentry/discover/translation/mep_to_eap.py

```diff
@@ -0,0 +1,74 @@
+from typing import TypedDict
+
+from parsimonious import NodeVisitor
+
+from sentry.api.event_search import event_search_grammar
+
+
+def apply_is_segment_condition(query: str) -> str:
+    if query:
+        return f"({query}) AND is_transaction:1"
+    return "is_transaction:1"
+
+
+def switcheroo(term):
+    if term == "transaction.duration":
+        return "span.duration"
+
+    return term
+
+
+class TranslationVisitor(NodeVisitor):
+    def __init__(self):
+        super().__init__()
+
+    def visit_raw_aggregate_param(self, node, children):
+        return switcheroo(node.text)
+
+    def visit_key(self, node, children):
+        return switcheroo(node.text)
+
+    def visit_value(self, node, children):
+        return switcheroo(node.text)
+
+    def generic_visit(self, node, children):
+        return children or node.text
+
+
+class QueryParts(TypedDict):
+    selected_columns: list[str]
+    query: str
+    equations: list[str] | None
+    orderby: list[str] | None
+
+
+# This is a utility used to translate transactions/metrics/mep
+# queries to eap queries. It takes in event query syntax (EQS)
+# as input and outputs EQS as well. This will allow us to
+# translate transaction queries from the frontend on the fly
+# and also allow us to migrate all our Discover/Dashboard/Alert
+# datamodels to store EAP compatible EQS queries.
+def translate_mep_to_eap(query_parts: QueryParts):
```

Comment:
nit
```suggestion
def translate_mep_to_eap(query_parts: QueryParts):
    """This is a utility used to translate transactions/metrics/mep
    queries to eap queries. It takes in event query syntax (EQS)
    as input and outputs EQS as well. This will allow us to
    translate transaction queries from the frontend on the fly
    and also allow us to migrate all our Discover/Dashboard/Alert
    datamodels to store EAP compatible EQS queries."""
```